Bump Cluster Autoscaler version to 1.17.0

losipiuk opened the pull request:
This PR updates CA version in gce manifests to 1.17.0

As every release cycle we are merging the change just before release. The reason for late update is fact that Cluster Autoscaler vendors k8s code to simulate scheduler behavior. Yet it lives in separate repository. Therefore we need to build final release candidate just at the end of k8s release cycle.

This PR contains also commit from https://github.com/kubernetes/kubernetes/pull/77895 which allows CA to read `lease` objects as we changed leader election to use those in 1.17.

/kind bug
/priority critical-urgent
/sig autoscaling
/assign @mwielgus

```release-note
Update Cluster Autoscaler to 1.17.0; changelog: https://github.com/kubernetes/autoscaler/releases/tag/cluster-autoscaler-1.17.0
```

Commit:
Make cluster auto scaler use leases

losipiuk committed:
Bump Cluster Autoscaler version to 1.17.0

losipiuk commented:
/hold

losipiuk commented:
/assign @liggitt

liggitt commented on cluster/addons/rbac/cluster-autoscaler/cluster-autoscaler-rbac.yaml:
```diff
@@ -6,6 +6,14 @@ metadata:
+    resources: ["leases"]
+    resourceNames: ["cluster-autoscaler"]
+    verbs: ["get", "update", "patch", "delete"]
+  # TODO: remove N releases after upgrading cluster autoscaler to a version that uses leases
```
can you add a name or specific release to this TODO? as-is, this isn't very actionable

liggitt commented:
one comment on the role, lgtm otherwise

MaciekPytel commented:
/retest

losipiuk commented:
Presubmit showed regression - we are investigating, and will update image when fixed.

losipiuk commented:
Updated to cluster-autoscaler-1.17-beta.2.
Let's wait for pull-kubernetes-e2e-gci-gce-autoscaling to complete.

losipiuk commented:
Presubmit passed:
https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/85610/pull-kubernetes-e2e-gci-gce-autoscaling/1200066560538972160/

There is one flake on `[sig-autoscaling] Cluster size autoscaling [Slow] should be able to scale down when rescheduling a pod is required and pdb allows for it[Feature:ClusterSizeAutoscalingScaleDown` yet it passed on retry.

I will already build `cluster-autoscaler-1.17.0` which will not contain any changes vs `-beta.2`. I will update this PR to use `-1.17.0` as otherwise we may run out of time before cherry-pick deadline on Monday.

losipiuk commented:
@liggitt I updated this PR to use `k8s.gcr.io/cluster-autoscaler:v1.17.0` (which has no changes vs 1.17.0-beta.2), and updated PR desc to include release notes.
I addressed your comment in `cluster-autoscaler-rbac.yaml`.

Unless you see any new issues with this one plese LGTM so we still have a change to request cherry-pick to `releases-1.17` branch

losipiuk commented:
/test pull-kubernetes-e2e-gci-gce-autoscaling

losipiuk commented:
/test pull-kubernetes-e2e-gce-100-performance

liggitt commented:
/lgtm
/approve
/hold for green autoscaling CI run, unhold once green

losipiuk commented:
One test failed. Flake as I did not see it fail before. Will run again though.

/test pull-kubernetes-e2e-gci-gce-autoscaling

losipiuk commented:
/test pull-kubernetes-e2e-gci-gce-autoscaling

losipiuk commented:
pull-kubernetes-e2e-gci-gce-autoscaling flaked again here: "https://github.com/kubernetes/kubernetes/blame/62965b9e47dabfb7c736027938fb23200a528914/test/utils/runners.go#L835"
restarting

losipiuk commented:
More flakes like before
https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/85610/pull-kubernetes-e2e-gci-gce-autoscaling/1200569641429110784/

/test pull-kubernetes-e2e-gci-gce-autoscaling

losipiuk commented:
Green !!!!
/hold cancel

losipiuk commented:
/test pull-kubernetes-e2e-gce
